src/Enhavo/Bundle/MediaBundle/Service/FileService.php

```diff
@@ -0,0 +1,357 @@
+     * @param bool                                  $garbage        [optional] Initial mark for garbage collector
+     * @throws \Exception   Thrown if file does not exist, is not readable, cannot be copied or parameter $file is of
+     *                      unknown type.
+     * @return EnhavoFile   The generated doctrine entity object (already persisted).
+     */
+    public function addFileToDatabase($file, $mimeType = null, $fileExtension = null, $title = null, $fileName = null, $order = null, $garbage = false)
+    {
+        $fileInfo = $this->getFileInformation($file);
+        if (!$fileInfo) {
+            throw new \Exception("EnhavoMediaBundle FileService Error: Invalid format on file parameter");
+        }
+
+        $slugifier = new Slugifier;
+
+        if ($mimeType == null) {
+            $mimeType = $fileInfo['mime_type'];
+        }
+        if ($fileExtension == null) {
+            $fileExtension = $fileInfo['extension'];
+        }
+        if ($title == null) {
+            $title = $fileInfo['filename'];
+        }
+        $title = $slugifier->slugify($title);
+        if ($fileName == null) {
+            $fileName = $fileInfo['basename'];
+        }
+
+        $entityFile = new EnhavoFile();
+        $entityFile->setMimeType($mimeType);
+        $entityFile->setExtension($fileExtension);
+        $entityFile->setTitle($title);
+        $entityFile->setFilename($fileName);
+        $entityFile->setOrder($order);
+        $entityFile->setGarbage($garbage);
+
+        $this->manager->persist($entityFile);
+        $this->manager->flush();
+
+        try {
+            copy($fileInfo['pathname'], $this->getDirectory($entityFile) . '/' . $entityFile->getId());
+        } catch(\Exception $exception) {
+            $this->manager->remove($entityFile);
+            $this->manager->flush();
+            throw $exception;
+        }
+
+        return $entityFile;
+    }
+
+    /**
+     * Used by addFileToDatabase() to wrap the different possibilities for types of the parameter $file.
+     * For a list of possible types, see addFileToDatabase().
+     *
+     * Returns an array containing:
+     *  'pathname'  => The full name of the file (including path)
+     *  'extension' => The file extension
+     *  'basename'  => The name of the file (with extension, without path)
+     *  'filename'  => The name of the file (without path or extension)
+     *  'mime_type' => The mime type for the file
+     *
+     * Returns null if $file is of no known type.
+     *
+     * @param string|\SplFileInfo|File|UploadedFile $file
+     * @return array|null
+     * @throws \Exception Thrown of the file is not found or not readable for apache user.
+     */
+    protected function getFileInformation($file)
+    {
+        if (is_string($file)) {
+            if (!is_readable($file)) {
+                throw new \Exception("EnhavoMediaBundle FileService Error: Unable to add file \"$file\" to database, file not found or not readable.");
```

Comment:
"Unable to add file \"$file\" to database" is not part of this function. Just say "File not found or not readable". And Please don't throw \Exception use different or create a specified one.